### ISSUE TITLE
Semgrep does not detect java files under directory named test because of default .semgrepignore

### ISSUE THREAD

**Issue by timhemel** [semgrep/semgrep#4568]
**Describe the bug**
I am running semgrep on a directory with java files, but it fails to detect a java file that is under a directory called `test`.

**To Reproduce**
I used the java rules from the semgrep-rules repository.

```
$ git clone https://github.com/Netflix/conductor.git

$ semgrep -f ~/semgrep/semgrep-rules/java/ conductor/redis-persistence/src/test/java/com/netflix/conductor/redis/dao/RedisPollDataDAOTest.java
Running 97 rules...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|97/97
ran 97 rules on 0 files: 0 findings

$ semgrep -f ~/semgrep/semgrep-rules/java/ $(pwd)/conductor/redis-persistence/src/test/java/com/netflix/conductor/redis/dao/RedisPollDataDAOTest.java
Running 97 rules...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|97/97
ran 97 rules on 0 files: 0 findings

$ cd conductor/redis-persistence/src/test/java/com/netflix/conductor/redis/dao/
$ semgrep -f ~/semgrep/semgrep-rules/java/ RedisPollDataDAOTest.java
Running 97 rules...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|97/97
ran 97 rules on 1 files: 0 findings

$ semgrep -f ~/semgrep/semgrep-rules/java/ $(pwd)/RedisPollDataDAOTest.java
Running 97 rules...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|97/97
ran 97 rules on 1 files: 0 findings
```

So, if I am in the directory containing the file, it detects it as a Java file (`ran 97 rules on 1 files`), but if I am not, it does not detect it (`ran 97 rules on 0 files`), regardless of whether I specify the full absolute path or not. If we go up the directory tree, we see that it still works, until we go to the parent of the directory `test`:

```
dao$ cd ..
redis$ semgrep -f ~/semgrep/semgrep-rules/java/ $(pwd)/dao/RedisPollDataDAOTest.java
Running 97 rules...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|97/97
ran 97 rules on 1 files: 0 findings
redis$ semgrep -f ~/semgrep/semgrep-rules/java/ dao/RedisPollDataDAOTest.java
Running 97 rules...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|97/97
ran 97 rules on 1 files: 0 findings
redis$ cd ..
conductor$ semgrep -f ~/semgrep/semgrep-rules/java/ redis/dao/RedisPollDataDAOTest.java
Running 97 rules...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|97/97
ran 97 rules on 1 files: 0 findings
conductor$ cd ..
netflix$ semgrep -f ~/semgrep/semgrep-rules/java/ conductor/redis/dao/RedisPollDataDAOTest.java
Running 97 rules...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|97/97
ran 97 rules on 1 files: 0 findings
netflix$ cd ..
com$ semgrep -f ~/semgrep/semgrep-rules/java/ netflix/conductor/redis/dao/RedisPollDataDAOTest.java
Running 97 rules...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|97/97
ran 97 rules on 1 files: 0 findings
com$ cd ..
java$ semgrep -f ~/semgrep/semgrep-rules/java/ com/netflix/conductor/redis/dao/RedisPollDataDAOTest.java
Running 97 rules...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|97/97
ran 97 rules on 1 files: 0 findings
java$ cd ..
test$ semgrep -f ~/semgrep/semgrep-rules/java/ java/com/netflix/conductor/redis/dao/RedisPollDataDAOTest.java
Running 97 rules...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|97/97
ran 97 rules on 1 files: 0 findings
test$ cd ..
src$ semgrep -f ~/semgrep/semgrep-rules/java/ test/java/com/netflix/conductor/redis/dao/RedisPollDataDAOTest.java
Running 97 rules...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|97/97
ran 97 rules on 0 files: 0 findings
```

Hypothesis: if the path contains a component called `test`, semgrep skips it, at least for Java files.

**Expected behavior**

I expect semgrep to find java files, even if their path contains a component called `test`.

**Screenshots**

**What is the priority of the bug to you?**

- [ ] P0: blocking your adoption of Semgrep or workflow
- [x] P1: important to fix or quite annoying
- [ ] P2: regular bug that should get fixed

**Environment**

Local build:

```
$ semgrep --version
0.79.0
$ uname -a
Darwin XXXX 20.6.0 Darwin Kernel Version 20.6.0: Wed Nov 10 22:23:07 PST 2021; root:xnu-7195.141.14~1/RELEASE_X86_64 x86_64
$ python3.9 --version
Python 3.9.7
```

**Comment by timhemel** [comment]
Confirmation that it is indeed the name `test` that gives problems:

```
src$ mv test best
src$ semgrep -f ~/semgrep/semgrep-rules/java/ best/java/com/netflix/conductor/redis/dao/RedisPollDataDAOTest.java
Running 97 rules...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|97/97
ran 97 rules on 1 files: 0 findings
```

**Comment by timhemel** [comment]
I can reproduce the same problem for python files.

**Comment by timhemel** [comment]
Looks like the default `.semgrepignore` file in the installed python package skips these directories. While this may be user friendly in a way, it is also not obvious to users, and leads to surprises. If they serve as examples of how make a `.semgrepignore` file, please comment out any such entries. And make users aware that a default ignore file is used.

**Comment by ievans** [comment]
@timhemel your assessment is correct. We're working on a couple improvements here around both messaging and defaults, more details soon. Previously: https://github.com/returntocorp/semgrep/issues/1964

**Comment by ievans** [comment]
@underyx closing this as fixed by #4616 and the docs PRs made by @s-santillan
